spec/person_spec.rb

```diff
@@ -0,0 +1,25 @@
+require_relative 'spec_helper'
+
+describe Person do
+  context 'When testing the Person class' do
+    before :each do
+      @name = 'Jorge'
+      @age = 21
+      @parent_permission = false
+      @jorge = Person.new(@age, @name, parent_permission: @parent_permission)
+    end
+    it 'Checks to see if a variable is assigned to an instance of the Person class' do
+      @jorge.should be_an_instance_of Person
+    end
+
+    it 'Checks to see if the Person instance is properly assigned the name of Jorge' do
+      actual = 'Jorge'
+      expect(@jorge.name).to eq actual
+    end
+
+    it 'Checks to see if the person object returns its correct name using the correct_name method' do
+      actual = 'Jorge'
+      expect(@jorge.correct_name).to eq actual
+    end
+  end
```

Comment:
- [ ] Great job with test coverage so far, but since we are aiming for a 100% test coverage, it will be nice to add tests for all our methods in all classes, for example for this `Person class`, we could add the following tests; 
- Test for adding a new rental method, something like so:
```
  context "When adding a new rentals" do

    it "The add_rental method returns a rental and adds its to the book" do
      book = Book.new("Title", "Author")
      person = Person.new(24, "Julius")

      rental = person.add_rental(book, "2021-05-04")

      expect(person.rentals).to eql [rental]
    end
  end
```
- Test for checking permissions 
```ruby
  context "When checking permissions" do
    
    it "The can_use_services? method returns false if under age and no parent permission" do
      person = Person.new(14, "Julius", false)
      expect(person.can_use_services?).to be false
    end
    
    it "The can_use_services? method returns true if over age" do
      person = Person.new(19, "Mihndim", false)
      expect(person.can_use_services?).to be true
    end
    
    it "The can_use_services? method returns true if has parent permission" do
      person = Person.new(14, "Rose", true)
      expect(person.can_use_services?).to be true
    end
  end
```
- Test for validating a name
```ruby
  context "When validating name" do

    it "The validate_name method corrects the name" do
      person = Person.new(19, "alongnameiscorrected")
      person.validate_name()
      expect(person.name).to eql "Alongnamei"
    end
  end
```
- Check if there are  some positive and negative case scenarios.

- [ ] As it is a [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_unit_tests.md) for this project, " `Add unit tests to your classes: all classes that you created while implementing the Decorator design pattern ` ", please kindly add tests for the decorator classes. 
### Please let's make sure all the methods in all the classes is covered with test.